x-pack/plugins/enterprise_search/public/applications/app_search/components/schema/schema_logic.ts

```diff
@@ -0,0 +1,170 @@
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { kea, MakeLogicType } from 'kea';
+import { isEqual } from 'lodash';
+
+import {
+  flashAPIErrors,
+  setErrorMessage,
+  flashSuccessToast,
+  clearFlashMessages,
+} from '../../../shared/flash_messages';
+import { HttpLogic } from '../../../shared/http';
+import { Schema, SchemaType, IndexJob } from '../../../shared/schema/types';
+import { EngineLogic } from '../engine';
+
+import { ADD_SCHEMA_ERROR, ADD_SCHEMA_SUCCESS, UPDATE_SCHEMA_SUCCESS } from './constants';
+import { SchemaBaseLogic, SchemaBaseValues, SchemaBaseActions } from './schema_base_logic';
+import { SchemaApiResponse } from './types';
+
+interface SchemaValues extends SchemaBaseValues {
+  isUpdating: boolean;
+  hasSchema: boolean;
+  hasSchemaChanged: boolean;
+  cachedSchema: Schema;
+  mostRecentIndexJob: Partial<IndexJob>;
+  unconfirmedFields: string[];
+  hasUnconfirmedFields: boolean;
+  hasNewUnsearchedFields: boolean;
+  isModalOpen: boolean;
+}
+
+interface SchemaActions extends SchemaBaseActions {
+  loadIndexedEngineSchema(): void;
+  onSchemaLoad(response: SchemaApiResponse): SchemaApiResponse;
+  addSchemaField(
+    fieldName: string,
+    fieldType: SchemaType
+  ): { fieldName: string; fieldType: SchemaType };
+  updateSchemaFieldType(
+    fieldName: string,
+    fieldType: SchemaType
+  ): { fieldName: string; fieldType: SchemaType };
+  updateSchema(successMessage?: string): string | undefined;
+  onSchemaUpdateError(): void;
+  openModal(): void;
+  closeModal(): void;
+}
+
+export const SchemaLogic = kea<MakeLogicType<SchemaValues, SchemaActions>>({
+  path: ['enterprise_search', 'app_search', 'schema_logic'],
+  connect: {
+    values: [SchemaBaseLogic, ['dataLoading', 'schema']],
+    actions: [SchemaBaseLogic, ['loadSchema', 'setSchema']],
+  },
+  actions: {
+    loadIndexedEngineSchema: true,
+    onSchemaLoad: (response) => response,
+    addSchemaField: (fieldName, fieldType) => ({ fieldName, fieldType }),
+    updateSchemaFieldType: (fieldName, fieldType) => ({ fieldName, fieldType }),
+    updateSchema: (successMessage) => successMessage,
+    onSchemaUpdateError: true,
+    openModal: true,
+    closeModal: true,
+  },
+  reducers: {
+    isUpdating: [
+      false,
+      {
+        updateSchema: () => true,
+        onSchemaLoad: () => false,
+        onSchemaUpdateError: () => false,
+      },
+    ],
+    cachedSchema: [
+      {},
+      {
+        onSchemaLoad: (_, { schema }) => schema,
+      },
+    ],
+    mostRecentIndexJob: [
+      {},
+      {
+        onSchemaLoad: (_, { mostRecentIndexJob }) => mostRecentIndexJob,
+      },
+    ],
+    unconfirmedFields: [
+      [],
+      {
+        onSchemaLoad: (_, { unconfirmedFields }) => unconfirmedFields,
+      },
+    ],
+    hasNewUnsearchedFields: [
+      false,
+      {
+        onSchemaLoad: (_, { unsearchedUnconfirmedFields }) => unsearchedUnconfirmedFields,
```

Comment:
FYI that I deliberately renamed this bool/flag name. `unsearchedUnconfirmedFields` does not read like a bool/flag to me, it reads like a list of fields 🤷 I remember getting really confused by that some time ago when fixing some schema callouts.

Hoping `hasNewUnsearchedFields` reads more clearly as a flag to others, as well as what it's flagging.